### PR TITLE
Lint for 'code' field on enum details.

### PR DESCRIPTION
**Pull request by stephenh**

### CODE REVIEW

File: integration/queries.graphql [hunk 1]
```diff
@@ -1,5 +1,8 @@
 query LoadAuthors {
   authors {
     name
+    status {
+      name
```
Comment (stephenh) [inline integration/queries.graphql:5]: For now the "test" is just running `npm run graphql-codegen` locally and watching for the error to get reported...

File: src/IdOnObjectSelectionSetRule.ts [hunk 2]
```diff
@@ -68,3 +82,18 @@ function selectionSetIncludesId(selectionNodes: readonly SelectionNode[]) {
...
+
+/**
+ * Look for the FooDetail/code/name pattern of our enum detail objects.
+ *
+ * NOTE: Borrowed from graphql-typescript-factories. A helper library may be in order?
+ */
+function isEnumDetailObject(object: GraphQLNamedType): object is GraphQLObjectType {
+  return (
+    object instanceof GraphQLObjectType &&
+    object.name.endsWith("Detail") &&
```
Comment (bdow) [inline src/IdOnObjectSelectionSetRule.ts:94]: Are we ensuring somewhere that all of our enum types end with "Detail"?

Comment (stephenh) [reply src/IdOnObjectSelectionSetRule.ts:94]: Yeah, it's part of the joist codegen infra on the `graphql-service` side of things.

File: integration/queries.graphql (hunk 1 above)
Comment (bdow) [reply integration/queries.graphql:5]: Hmm.. I was about to ask for a test regarding this. I guess this will suffice for now.

**Review by bdow — APPROVED** [review]

File: src/IdOnObjectSelectionSetRule.ts (hunk 2 above)
Comment (chr1sjf0x) [reply src/IdOnObjectSelectionSetRule.ts:94]: Yes, we need to move this helper into it's own repo by now 😆

**Review by chr1sjf0x — APPROVED** [review]
TY @stephenh

**Pull request merged**